src/server/multi_test.cc

```diff
@@ -756,4 +727,61 @@ TEST_F(MultiTest, TestSquashing) {
   f1.Join();
 }
 
+class MultiEvalTest : public BaseFamilyTest {
+ protected:
+  MultiEvalTest() : BaseFamilyTest() {
+    num_threads_ = kPoolThreadCount;
+    absl::SetFlag(&FLAGS_default_lua_flags, "allow-undeclared-keys");
+  }
+
+  absl::FlagSaver fs_;
+};
+
+TEST_F(MultiEvalTest, MultiAllEval) {
+  if (auto mode = absl::GetFlag(FLAGS_multi_exec_mode); mode == Transaction::NON_ATOMIC) {
+    GTEST_SKIP() << "Skipped MultiAllEval test because multi_exec_mode is non atomic";
+    return;
+  }
+
+  RespExpr brpop_resp;
+
+  // Run the fiber at creation.
+  auto fb0 = pp_->at(1)->LaunchFiber(Launch::dispatch, [&] {
+    brpop_resp = Run({"brpop", "x", "1"});
+  });
+  Run({"multi"});
+  Run({"eval", "return redis.call('lpush', 'x', 'y')", "0"});
+  Run({"eval", "return redis.call('lpop', 'x')", "0"});
+  RespExpr exec_resp = Run({"exec"});
+  fb0.Join();
+
+  ASSERT_THAT(exec_resp, ArrLen(2));
+  ASSERT_THAT(exec_resp.GetVec(), ElementsAre(IntArg(1), "y"));
```

Comment:
Nit: You could replace these 2 lines with `EXPECT_THAT(exec_rep, RespArray(ElementsAre(IntArg(1), "y"));`
BTW why `ASSERT` and not `EXPECT`?